src/eckit/parser/YAMLParser.cc

```diff
@@ -443,27 +479,30 @@ static Value toValue(const std::string& s) {
     if (s.length()) {
         // This is because checking regex is very slow
         switch (s[0]) {
+            case '-':
+            case '+':
             case '0':
 
-                if (octal.match(s)) {
-                    return Value(strtol(s.c_str(), 0, 0));
+                if (integer2.match(s)) {
+                    std::string str = ignoreChar(s, '_');
```

Comment:
Typically ignoreChars will be a nop (most of the numbers parsed will not be containing '_'). This introduces an annoying memory copy here. Is it possible to avoid creating a new string except in cases where it is needed?

Reply:
I have made a change to `ignoreChar`. Let me know if it does what you want here or not.